io.github.ja2_stracciatella.JA2-Stracciatella.yaml

```diff
@@ -0,0 +1,141 @@
+      - "-DSOL2_BUILD_LUA=OFF"
+      - "-DSOL2_TESTS_SINGLE=OFF"
+      - "-DSOL2_TESTS_EXAMPLES=OFF"
+      - "-Wno-dev"
+    sources:
+      - type: archive
+        url: "https://github.com/ThePhD/sol2/archive/refs/tags/v3.2.2.tar.gz"
+        sha256: "141790dae0c1821dd2dbac3595433de49ba72545845efc3ec7d88de8b0a3b2da"
+    cleanup:
+      - "/include"
+      - "/share/doc"
+      - "/share/pkgconfig"
+      - "/lib/cmake"
+  - name: ja2-stracciatella
+    buildsystem: cmake-ninja
+    build-options:
+      env:
+        # Point cargo to crates downloaded by flatpak from cargo-sources.json
+        CARGO_HOME: "/run/build/ja2-stracciatella/cargo"
+      # Add cargo to $PATH while building this module
+      append-path: "/usr/lib/sdk/rust-stable/bin"
+    config-opts:
+      - "-DEXTRA_DATA_DIR=/app/share/ja2"
+      - "-DLOCAL_LUA_LIB=OFF"
+      - "-DLOCAL_SOL_LIB=OFF"
+      - "-DLOCAL_RAPIDJSON_LIB=OFF"
+      - "-DLOCAL_STRING_THEORY_LIB=OFF"
+      - "-DLOCAL_MINIAUDIO_LIB=OFF"
+      - "-DMINIAUDIO_INCLUDE_DIR=/run/build/ja2-stracciatella/dependencies/lib-miniaudio/src"
+      - "-DLOCAL_MAGICENUM_LIB=OFF"
+      - "-DMAGICENUM_INCLUDE_DIR=/run/build/ja2-stracciatella/dependencies/lib-magic_enum/src"
+      - "-DWITH_UNITTESTS=OFF"
+      - "-Wno-dev"
+    sources:
+      - type: archive
+        url: https://github.com/ja2-stracciatella/ja2-stracciatella/archive/refs/tags/v0.21.0.tar.gz
+        sha256: "1c15b4f281bba9aff1c3409d4308d22840eb815a8a3ab53f7be69a5841f300a2"
+      # Cargo sources, generated with
+      #
+      #   python3 flatpak-cargo-generator.py JA2_CHECKOUT/rust/Cargo.lock -o cargo-sources.json
+      #
+      # See https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo
+      # for details.
+      - cargo-sources.json
+      # Directly download required header-only or direct-include libraries.
+      # Use exactly the same versions as used by JA2 upstream, in the CMake
+      # getters.
+      - type: archive
+        url: "https://github.com/mackron/miniaudio/archive/4dfe7c4c31df46e78d9a1cc0d2d6f1aef5a5d58c.tar.gz"
+        sha256: "76c154a60e320ae2054ac0e93480f2dffc12a5129bdb2ed4a62e0cce8d345c36"
+        dest: "dependencies/lib-miniaudio/src"
+      - type: file
+        url: https://github.com/Neargye/magic_enum/releases/download/v0.8.2/magic_enum.hpp
+        sha256: "f34487663db05b10acae7077dd0b5cf5794112a379567322e251ef0068875dc4"
+        dest: "dependencies/lib-magic_enum/src"
+    cleanup:
+      - /share/ja2/unittests
+      - /share/man
+  # Use a separate module for the launcher script and extra metadata to avoid
+  # rebuilding JA2 when changing these.
+  - name: ja2-flatpak
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 ja2-launcher-wrapper /app/bin/ja2-launcher-wrapper
+      - install -Dm644 io.github.ja2_stracciatella.JA2-Stracciatella.appdata.xml "${FLATPAK_DEST}/share/metainfo/io.github.ja2_stracciatella.JA2-Stracciatella.metainfo.xml"
+    sources:
```

Comment:
```suggestion
    cleanup:
      - /share/ja2/unittests
      - /share/man
    post-install:
      - install -Dm755 ja2-launcher-wrapper /app/bin/ja2-launcher-wrapper
      - install -Dm644 io.github.ja2_stracciatella.JA2-Stracciatella.appdata.xml "${FLATPAK_DEST}/share/metainfo/io.github.ja2_stracciatella.JA2-Stracciatella.metainfo.xml"
    sources:
      - type: archive
        url: https://github.com/ja2-stracciatella/ja2-stracciatella/archive/refs/tags/v0.21.0.tar.gz
        sha256: "1c15b4f281bba9aff1c3409d4308d22840eb815a8a3ab53f7be69a5841f300a2"
      # Cargo sources, generated with
      #
      #   python3 flatpak-cargo-generator.py JA2_CHECKOUT/rust/Cargo.lock -o cargo-sources.json
      #
      # See https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo
      # for details.
      - cargo-sources.json
      # Directly download required header-only or direct-include libraries.
      # Use exactly the same versions as used by JA2 upstream, in the CMake
      # getters.
      - type: archive
        url: "https://github.com/mackron/miniaudio/archive/4dfe7c4c31df46e78d9a1cc0d2d6f1aef5a5d58c.tar.gz"
        sha256: "76c154a60e320ae2054ac0e93480f2dffc12a5129bdb2ed4a62e0cce8d345c36"
        dest: "dependencies/lib-miniaudio/src"
      - type: file
        url: https://github.com/Neargye/magic_enum/releases/download/v0.8.2/magic_enum.hpp
        sha256: "f34487663db05b10acae7077dd0b5cf5794112a379567322e251ef0068875dc4"
        dest: "dependencies/lib-magic_enum/src"
```

Reply:
As far as I can see this merges the start script module into the build module?  Doesn't this mean I have to rebuild the entire game (that's not exactly fast, rust isn't known for its fast compile times 😬) whenever I want to test modifications to the startup script?

Reply:
aren't you done testing them?

Reply:
> I figured it'd be good style to keep independent

it's not.

also the appstream file should be upstreamed for a future release.

Reply:
> it's not.

Mmh, this starts to feel a lot like my last attempt… 😒   I'd appreciate if you could give me something like an explanation at least…

> also the appstream file should be upstreamed for a future release.

I can try to upstream it, but I'd prefer if I could defer this until this manifest is merged.  I mean no offence, but if this goes nowhere like the last attempt did then I can just as well save myself some work 🙂